benchmark/cublas/blas2/ger.cpp

```diff
@@ -0,0 +1,191 @@
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename ger.cpp
+ *
+ **************************************************************************/
+
+#include "../utils.hpp"
+
+template <typename scalar_t>
+std::string get_name(std::string t, int m, int n) {
+  std::ostringstream str{};
+  str << "BM_Ger<" << blas_benchmark::utils::get_type_name<scalar_t>() << ">/"
+      << t << "/" << m << "/" << n;
+  return str.str();
+}
+
+template <typename scalar_t, typename... args_t>
+static inline void cublas_routine(args_t&&... args) {
+  if constexpr (std::is_same_v<scalar_t, float>) {
+    CUBLAS_CHECK(cublasSger(std::forward<args_t>(args)...));
+  } else if constexpr (std::is_same_v<scalar_t, double>) {
+    CUBLAS_CHECK(cublasDger(std::forward<args_t>(args)...));
+  }
+  return;
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, cublasHandle_t* cuda_handle_ptr, index_t m,
+         index_t n, scalar_t alpha, bool* success) {
+  // Standard test setup.
+
+  index_t xlen = m;
+  index_t ylen = n;
+
+  index_t lda = m;
+  index_t incX = 1;
+  index_t incY = 1;
+
+  // The counters are double. We convert m and n to double to avoid
+  // integer overflows for n_fl_ops and bytes_processed
+  double m_d = static_cast<double>(m);
+  double n_d = static_cast<double>(n);
+
+  state.counters["m"] = m_d;
+  state.counters["n"] = n_d;
+
+  {
+    double nflops_timesAlpha = std::min(m, n);
+    double nflops_XtimesYplusA = 2 * m_d * n_d;
+    state.counters["n_fl_ops"] = nflops_XtimesYplusA + nflops_timesAlpha;
+  }
+  {
+    double mem_readWriteA = 2 * m_d * n_d;
+    double mem_readX = xlen;
+    double mem_readY = ylen;
+    state.counters["bytes_processed"] =
+        (mem_readWriteA + mem_readX + mem_readY) * sizeof(scalar_t);
+  }
+
+  cublasHandle_t& cuda_handle = *cuda_handle_ptr;
+
+  // Input matrix/vector, output vector.
+  std::vector<scalar_t> m_a =
+      blas_benchmark::utils::random_data<scalar_t>(m * n);
+  std::vector<scalar_t> v_x =
+      blas_benchmark::utils::random_data<scalar_t>(xlen);
+  std::vector<scalar_t> v_y =
+      blas_benchmark::utils::random_data<scalar_t>(ylen);
+
+  cublasSetPointerMode(cuda_handle, CUBLAS_POINTER_MODE_HOST);
```

Comment:
Why do we need to set the `cublasSetPointerMode` here?

Reply:
We don't. Thank you for spotting it.